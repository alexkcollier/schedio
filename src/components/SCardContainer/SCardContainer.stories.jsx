import { SCardContainer } from '@/components/SCardContainer'
import { SHeading } from '@/components/SHeading'
import PropList from '@/docs/PropList'
import { text } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/vue'

storiesOf('Components/Layout.SCard', module)
  .addParameters({ jest: 'SCardContainer' })
  .add('SCardContainer', () => {
    return {
      props: {
        slotContent: { default: text('default', 'This is a SCardContainer!', 'Slots') }
      },
      render(h) {
        return (
          <div>
            <SHeading level="1">Card Container</SHeading>
            <p>Creates a container that also looks like a card.</p>

            <SHeading level="2">Example</SHeading>
            <SCardContainer>{this.slotContent}</SCardContainer>

            {SCardContainer.props && <PropList component={SCardContainer} />}
          </div>
        )
      }
    }
  })
