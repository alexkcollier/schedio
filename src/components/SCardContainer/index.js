import SCardContainer from './SCardContainer.vue'

export { SCardContainer }

export default { install: Vue => Vue.component(SCardContainer.name, SCardContainer) }
