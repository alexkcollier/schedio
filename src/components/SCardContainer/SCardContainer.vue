<template>
  <SContainer>
    <SCard :tag="tag">
      <slot />
    </SCard>
  </SContainer>
</template>

<script>
import { SCard } from '@/components/SCard'
import { SContainer } from '@/components/SContainer'

export default {
  name: 'SCardContainer',

  components: {
    SCard,
    SContainer,
  },

  props: {
    tag: {
      type: String,
      default: 'div',
    },
  },
}
</script>
