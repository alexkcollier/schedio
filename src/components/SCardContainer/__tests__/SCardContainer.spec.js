import { shallowMount } from '@vue/test-utils';
import SCardContainer from '@/components/SCardContainer/SCardContainer.vue';
import { axe, toHaveNoViolations } from 'jest-axe';

expect.extend(toHaveNoViolations);

describe('SCardContainer.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(SCardContainer, {
      slots: {
        default: '<div>Slot content</div>',
      },
    });
  });

  it('renders correctly', async () => {
    expect(wrapper.html()).toMatchSnapshot();
    expect(await axe(wrapper.html())).toHaveNoViolations();
  });

  it('can have content', () => {
    expect(wrapper.vm.$slots.default).toBeTruthy();
  });

  it('can be any html element', async () => {
    wrapper.setProps({ tag: 'section' });
    await wrapper.vm.$nextTick();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
