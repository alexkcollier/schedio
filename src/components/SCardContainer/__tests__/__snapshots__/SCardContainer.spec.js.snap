// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SCardContainer.vue can be any html element 1`] = `
<scontainer-stub tag="div">
  <scard-stub tag="section" image="[object Object]">
    <div>Slot content</div>
  </scard-stub>
</scontainer-stub>
`;

exports[`SCardContainer.vue renders correctly 1`] = `
<scontainer-stub tag="div">
  <scard-stub tag="div" image="[object Object]">
    <div>Slot content</div>
  </scard-stub>
</scontainer-stub>
`;
